packages/core/admin/admin/src/components/ThemeToggleProvider.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+import { DefaultTheme } from 'styled-components';
+
+import { ThemeToggleContext } from '../contexts/themeToggle';
+
+const THEME_KEY = 'STRAPI_THEME';
+
+const getDefaultTheme = () => {
+  const browserTheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+  const persistedTheme = localStorage.getItem(THEME_KEY) as 'dark' | 'light' | null;
+
+  if (!persistedTheme) {
+    localStorage.setItem(THEME_KEY, browserTheme);
+  }
+
+  return persistedTheme || browserTheme;
+};
+
+interface ThemeToggleProviderProps {
+  children: React.ReactNode;
+  themes: {
+    light: DefaultTheme;
+    dark: DefaultTheme;
```

Comment:
Should https://github.com/strapi/strapi/pull/18285/files#diff-e7a443022e5e7a0d9ec9c45ed6799c3c548df969213b2510b86df735262148f8R11-R12 also use these?

Reply:
yes, i've changed them